### ISSUE TITLE
Release 4.6.0 - Alpha 1 - E2E UX tests - macOs

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#18838]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.6.0-alpha1/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, or Docker) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the CICD team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the CICD team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Communicate these to QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/early-birds team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Quickstart](https://documentation-dev.wazuh.com/v4.6.0-alpha1/quickstart.html) | - | RHEL 7 x86_64 |
| Server | [Quickstart](https://documentation-dev.wazuh.com/v4.6.0-alpha1/quickstart.html) | - | RHEL 7 x86_64 |
| Dashboard | [Quickstart](https://documentation-dev.wazuh.com/v4.6.0-alpha1/quickstart.html) | - | RHEL 7 x86_64 |
| Agent | [Installing Wazuh agents](https://documentation-dev.wazuh.com/v4.6.0-alpha1/installation-guide/wazuh-agent/index.html) | - | macOS Ventura x86_64, macOS Ventura arm |

## Test description

Test that macOs data collection works out of the box. 
Test that macOs vulnerability works out of the box.
Test that macOs file integrity monitoring works out of the box.  

## Known issues

- https://github.com/wazuh/wazuh/issues/15580

## Conclusions

All tests have been executed and the results can be found here.:

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| 🟢  | [Indexer](https://github.com/wazuh/wazuh/issues/18838#issuecomment-1711598444) | - | - |
| 🟢  | [Server](https://github.com/wazuh/wazuh/issues/18838#issuecomment-1711598444) | - | -  |
| 🟢  | [Dashboard](https://github.com/wazuh/wazuh/issues/18838#issuecomment-1711598444) | - | - |
| 🟢  | [Agent (x86_64)](https://github.com/wazuh/wazuh/issues/18838#issuecomment-1713536092) | - | - |
| 🟢  | [Agent (arm)](https://github.com/wazuh/wazuh/issues/18838#issuecomment-1712091236) | - | - |

## Feedback
The testing guide was quite detailed, however, the term data collection used here can be changed to log data collection as this is what we have in the documentation. 

## Reviewers validation

The criteria for completing this task is based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [ ] @davidjiglesias
- [x] @wazuh/early-birds

**Comment by ace-109** [comment]
### Tests (macOS Ventura ARM)

The following tests were carried out on a macOS Ventura ARM endpoint.

#### macOS data collection. 🟢

I attempted a brute-force attack on the macOS machine, this was detected along with other events out of the box as seen in the screenshot below

![image](https://github.com/wazuh/wazuh/assets/41423652/e0a73adb-210c-473b-9103-ebd87976a200)

#### macOS Vulnerability detection. 🟢

I had to enable vulnerability detection on the Wazuh server by setting enabled to yes on the `ossec.conf` file of the manager

```
  <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>5m</interval>
    <min_full_scan_interval>6h</min_full_scan_interval>
    <run_on_start>yes</run_on_start>
```

I restarted the Wazuh manager after this. I checked the log files and I can see the manager has scanned the macOS Ventura ARM (002) endpoint for vulnerabilities. There's a warning that it couldn't get vulnerability data for the agent '000' OS which is RHEL 7 (wazuh manager).

```
[root@localhost wazuh]# [root@localhost wazuh]# tail -20 /var/ossec/logs/ossec.log
2023/09/08 09:59:43 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'National Vulnerability Database' database update.
2023/09/08 09:59:43 wazuh-modulesd:syscollector: INFO: Module started.
2023/09/08 09:59:43 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2023/09/08 09:59:44 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel7_linux.yml'
2023/09/08 09:59:49 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2023/09/08 09:59:55 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2023/09/08 09:59:55 wazuh-syscheckd: INFO: FIM sync module started.
2023/09/08 10:00:06 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel7_linux.yml'
2023/09/08 10:00:06 sca: INFO: Security Configuration Assessment scan finished. Duration: 23 seconds.
2023/09/08 10:00:48 rootcheck: INFO: Ending rootcheck scan.
2023/09/08 10:07:59 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'National Vulnerability Database' feed finished successfully.
2023/09/08 10:07:59 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Microsoft Security Update' database update.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Microsoft Security Update' feed finished successfully.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: WARNING: (5575): Unavailable vulnerability data for the agent '000' OS. Skipping it.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: INFO: (5450): Analyzing agent '001' vulnerabilities.
2023/09/08 10:09:12 wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment for agent '001'
2023/09/08 10:09:12 wazuh-modulesd:vulnerability-detector: INFO: (5450): Analyzing agent '002' vulnerabilities.
2023/09/08 10:09:24 wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment for agent '002'
2023/09/08 10:09:24 wazuh-modulesd:vulnerability-detector: INFO: (5472): Vulnerability scan finished.
[root@localhost wazuh]#
```
I checked the dashboard and it shows the vulnerabilities found on the macOS Ventura ARM endpoint 
![image](https://github.com/wazuh/wazuh/assets/41423652/52021139-37d9-4ec5-aa07-c4e3d595518c)

#### macOS File integrity monitoring. 🟢
macOS only supports scheduled scans, out of the box, this was scheduled to run every 12hours as seen in the config below

```
  <!-- File integrity monitoring -->
  <syscheck>
    <disabled>no</disabled>

    <!-- Frequency that syscheck is executed default every 12 hours -->
    <frequency>43200</frequency>
```

To reduce the waiting time, I modified the configuration to set FIM to run every 30 mins 
```
  <!-- File integrity monitoring -->
  <syscheck>
    <disabled>no</disabled>

    <!-- Frequency that syscheck is executed default every 12 hours -->
    <frequency>1800</frequency>
```

I restarted the macOS Ventura ARM agent

```
sh-3.2# /Library/Ossec/bin/wazuh-control restart
Killing wazuh-modulesd...
Killing wazuh-logcollector...
Killing wazuh-syscheckd...
Killing wazuh-agentd...
Killing wazuh-execd...
Wazuh v4.6.0 Stopped
Starting Wazuh v4.6.0...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
```

I waited for about 35 mins, then I see File Integrity Monitoring events on the Wazuh dashboard
![macOS ARM FIM](https://github.com/wazuh/wazuh/assets/41423652/faef33bb-e8c1-4d9e-8023-17e42596dfed)

In summary, all the tests carried out on the macOS Ventura arm endpoint were successful.

**Comment by ace-109** [comment]
### Tests (macOS Ventura x86_64) 🟢

The following tests were carried out on a macOS Ventura x86_64 endpoint. 

macOS data collection. 🟢

I attempted a brute-force attack on the macOS machine, this was detected along with other events out of the box as seen in the screenshot below

![mac x86 data collection](https://github.com/wazuh/wazuh/assets/41423652/81d1e76b-720b-4e91-b170-bddeb73b766e)

macOS Vulnerability detection. 🟢

macOS only supports scheduled scans, out of the box, this was scheduled to run every 12 hours as seen in the config below

```
  <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>5m</interval>
    <min_full_scan_interval>6h</min_full_scan_interval>
    <run_on_start>yes</run_on_start>
```

I restarted the Wazuh manager after this. I checked the log files and I can see the manager has scanned the macOS Ventura x86_64 (001) endpoint for vulnerabilities. There's a warning that it couldn't get vulnerability data for the agent '000' OS which is RHEL 7 (wazuh manager).

```
[root@localhost wazuh]# [root@localhost wazuh]# tail -20 /var/ossec/logs/ossec.log
2023/09/08 09:59:43 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'National Vulnerability Database' database update.
2023/09/08 09:59:43 wazuh-modulesd:syscollector: INFO: Module started.
2023/09/08 09:59:43 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2023/09/08 09:59:44 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel7_linux.yml'
2023/09/08 09:59:49 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2023/09/08 09:59:55 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2023/09/08 09:59:55 wazuh-syscheckd: INFO: FIM sync module started.
2023/09/08 10:00:06 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel7_linux.yml'
2023/09/08 10:00:06 sca: INFO: Security Configuration Assessment scan finished. Duration: 23 seconds.
2023/09/08 10:00:48 rootcheck: INFO: Ending rootcheck scan.
2023/09/08 10:07:59 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'National Vulnerability Database' feed finished successfully.
2023/09/08 10:07:59 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Microsoft Security Update' database update.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Microsoft Security Update' feed finished successfully.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: WARNING: (5575): Unavailable vulnerability data for the agent '000' OS. Skipping it.
2023/09/08 10:08:22 wazuh-modulesd:vulnerability-detector: INFO: (5450): Analyzing agent '001' vulnerabilities.
2023/09/08 10:09:12 wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment for agent '001'
2023/09/08 10:09:12 wazuh-modulesd:vulnerability-detector: INFO: (5450): Analyzing agent '002' vulnerabilities.
2023/09/08 10:09:24 wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment for agent '002'
2023/09/08 10:09:24 wazuh-modulesd:vulnerability-detector: INFO: (5472): Vulnerability scan finished.
[root@localhost wazuh]#
```

I checked the dashboard and it shows the vulnerabilities found on the macOS Ventura x86_64 endpoint

![image](https://github.com/wazuh/wazuh/assets/41423652/cc05945a-d23e-4182-b480-b33c9bfe036e)

macOS File integrity monitoring. 🟢

macOS only supports scheduled scans, out of the box, this was scheduled to run every 12 hours as seen in the config below

```
  <!-- File integrity monitoring -->
  <syscheck>
    <disabled>no</disabled>

    <!-- Frequency that syscheck is executed default every 12 hours -->
    <frequency>43200</frequency>
```
I modified the configuration to set FIM to run every 30 mins

```
  <!-- File integrity monitoring -->
  <syscheck>
    <disabled>no</disabled>

    <!-- Frequency that syscheck is executed default every 12 hours -->
    <frequency>1800</frequency>
```

I restarted the Wazuh service to apply the new configuration

```
sh-3.2# /Library/Ossec/bin/wazuh-control restart
Killing wazuh-modulesd...
Killing wazuh-logcollector...
Killing wazuh-syscheckd...
Killing wazuh-agentd...
Killing wazuh-execd...
Wazuh v4.6.0 Stopped
Starting Wazuh v4.6.0...
Started wazuh-execd...
Started wazuh-agentd...
Started wazuh-syscheckd...
Started wazuh-logcollector...
Started wazuh-modulesd...
Completed.
```

I waited for about 35 mins and the File integrity monitoring scan can be seen on the Wazuh dashboard

![image](https://github.com/wazuh/wazuh/assets/41423652/919063cd-0fde-49c0-84a1-7a8589d6e1b6)

All the tests carried out on the macOS Ventura x86_64 were successful .

**Comment by nmkoremblum** [comment]
LGTM :+1:

**Comment by davidjiglesias** [comment]
LGTM!
